username_api.py

```diff
@@ -12,12 +14,57 @@
 
 patterns = yaml.load(open('websites.yml'))
 
-def check_username(website, username):
-	url = patterns['urls'].get(website, 'https://{w}.com/{u}').format(
+def get_profile_url(website, username):
+	return patterns['urls'].get(website, 'https://{w}.com/{u}').format(
 		w=website,
 		u=username
 	)
 
+def get_avatar(website, username):
+	data = patterns['avatar'][website]
+
+	if not data:
+		return None
+
+	url = get_profile_url(website, username)
+	if 'url' in data:
+		url = data['url'].format(u=username)
+
+	response = r.get(url)
+	if response.status_code == 404:
+		return None
+
+	if data == 'opengraph':
+		# Look in metadata for image.
+		soup = BeautifulSoup(response.text, 'html.parser')
+		result = [item.attrs['content'] for item in soup('meta')
+				  if item.has_attr('property') and
+				  item.attrs['property'].lower() == 'og:image']
+		if not result or not result[0]:
```

Comment:
why is check for both ``result`` and ``result[0]`` needed here?

Reply:
Yes, some website returned an empty string for some reason. This is for that. (I can't remember which one, I think it was fb)